src/Sharpliner/AzureDevOps/Model/Tasks/Marketplace/NuGetPackCommandTask.cs

```diff
@@ -0,0 +1,298 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using YamlDotNet.Serialization;
+
+namespace Sharpliner.AzureDevOps.Tasks
+{
+    /// <summary>
+    /// Represents the NuGetCommand@2 task for packing NuGet packages in Azure DevOps pipelines.
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// var packTask = new NuGetPackCommandTask
+    /// {
+    ///     PackagesToPack = "**/*.csproj",
+    ///     Arguments = "-Properties Configuration=Release"
+    /// };
+    /// </code>
+    /// <para>The corresponding YAML will be:</para>
+    /// <code>
+    /// - task: NuGetCommand@2
+    ///   inputs:
+    ///     command: pack
+    ///     packagesToPack: '**/*.csproj'
+    ///     arguments: '-Properties Configuration=Release'
+    /// </code>
+    /// </example>
+    public abstract record NuGetPackCommandTask : NuGetCommandTask
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NuGetPackCommandTask"/> class.
+        /// </summary>
+        public NuGetPackCommandTask(string versioningScheme) : base("pack")
+        {
+            VersioningScheme = versioningScheme;
+        }
+
+        /// <summary>
+        /// Gets or sets the pattern to search for csproj or nuspec files to pack.
+        /// </summary>
+        [YamlIgnore]
+        public string PackagesToPack
+        {
+            get => GetString("packagesToPack")!;
+            init => SetProperty("packagesToPack", value);
+        }
```

Comment:
This one is not required in the ctor so should be like this?

```suggestion
        public string? PackagesToPack
        {
            get => GetString("packagesToPack");
            init => SetProperty("packagesToPack", value);
        }
```